components/core/src/fs.rs

```diff
@@ -208,7 +256,208 @@ pub fn find_command(command: &str) -> Option<PathBuf> {
+        let new_path = env::join_paths(path_bufs).unwrap();
+        env::set_var("PATHEXT", &new_path);
+    }
+
+    fn setup_empty_pathext() {
+        match env::var("PATHEXT") {
+            Ok(val) => env::remove_var("PATHEXT"),
+            Err(e) => {}
+        }
+    }
+
+    fn setup_path() {
+        let first_path = fs::canonicalize("./tests/fixtures").unwrap();
+        let second_path = fs::canonicalize("./tests/fixtures/bin").unwrap();
+        let path_bufs = vec![first_path, second_path];
+        let new_path = env::join_paths(path_bufs).unwrap();
+        env::set_var("PATH", &new_path);
+    }
+
+    mod without_pathext_set {
+        use super::{setup_path, setup_empty_pathext};
+        pub use super::find_command;
+
+        fn setup_environment() {
+            setup_path();
+            setup_empty_pathext();
+        }
+
+        mod argument_without_extension {
+            use super::{setup_environment, find_command};
+
+            #[test]
+            fn command_exists() {
+                setup_environment();
+                let result = find_command("bin_with_no_extension");
+                assert_eq!(result.is_some(), true);
+            }
+
+            #[test]
+            fn command_does_not_exist() {
+                setup_environment();
+                let result = find_command("missing");
+                assert_eq!(result.is_some(), false);
+            }
+
+            #[test]
+            fn command_exists_with_extension() {
+                setup_environment();
+                let result = find_command("win95_dominator");
+                assert_eq!(result.is_some(), false);
+            }
+        }
+
+        mod argument_with_extension {
+            use std::fs::canonicalize;
+            use super::{setup_environment, find_command};
+
+            #[test]
+            fn command_exists() {
+                setup_environment();
+                let result = find_command("bin_with_extension.exe");
+                assert_eq!(result.is_some(), true);
+            }
+
+            #[test]
+            fn command_does_not_exist() {
+                setup_environment();
+                let result = find_command("missing.com");
+                assert_eq!(result.is_some(), false);
+            }
+
+            #[test]
+            fn command_different_extension_does_exist() {
+                setup_environment();
+                let result = find_command("bin_with_extension.com");
+                assert_eq!(result.is_some(), false);
+            }
+
+            #[test]
+            fn first_command_on_path_found() {
+                setup_environment();
+                let target_path = canonicalize("./tests/fixtures/plan.sh").unwrap();
+                let result = find_command("plan.sh");
+                let found_path = result.unwrap();
+                assert_eq!(found_path, target_path);
+            }
+        }
+    }
+
+    #[cfg(target_os="windows")]
```

Comment:
I started using the modules as a way to group the tests to keep the function names shorter, but it became handy when I needed to make one windows only :)